src/patient-search/patient-search-result.test.tsx

```diff
@@ -0,0 +1,83 @@
+import { render, fireEvent, wait, cleanup } from "@testing-library/react";
+import React from "react";
+import PatientSearch from "./patient-search.component";
+import { performPatientSearch } from "./patient-search.resource";
+import { debounce } from "lodash";
+
+const mockSearch = performPatientSearch as jest.Mock;
+const history = { goBack: () => {} };
+const patients = {
+  data: {
+    results: [
+      {
+        display: "94004EH: John Doe",
+        uuid: "8673ee4f-e2ab-4077-ba55-4980f408773e",
+        person: {
+          age: 32,
+          birthdate: "01-01-2009",
+          gender: "M"
+        },
+        identifiers: [
+          {
+            identifier: "94004EH",
+            identifierType: {
+              display: "OpenMRS ID"
+            }
+          }
+        ]
+      }
+    ]
+  }
+};
+
+jest.mock("./patient-search.resource", () => ({
+  performPatientSearch: jest.fn().mockResolvedValue({
+    data: {
+      results: []
+    }
+  })
+}));
+
+jest.mock("lodash", () => ({
+  debounce: jest.fn(fn => fn)
+}));
+
+afterEach(cleanup);
+
+describe("<PatientSearch/>", () => {
+  it("renders without failing", () => {
+    render(<PatientSearch history={history} />);
+  });
+
+  it("triggers search on typing", () => {
```

Comment:
Good job diving into testing. It's hard work and a sign of high quality code.

Here's a suggestion -- make your tests verify something that the user can see instead of verifying that the code has been called. For example, mock the API and then verify that there are patient search links for each mocked result.

This might be hard to do so don't worry about it too much, but that's one of the main ideas with testing uis -- your assertions should be based on what the user sees more than on whether the code was called.

Reply:
Thanks for the insight Joel